### PR TITLE
Add reverseStr and groupBy

### PR DESCRIPTION
**Pull request by rafaelss95**
I had to put the `reverseStr` name because there's already a pipe (for arrays) called `Reverse`, so it may be ambiguous.

### CODE REVIEW

**Comment by fknop** [comment]
Thanks,

You're missing a small test to reach 100% coverage. https://coveralls.io/builds/9868294/source?filename=src%2Fstring%2Freverse-str.pipe.ts

File: docs/string.md [hunk 1]
```diff
@@ -400,3 +401,19 @@ import { WithPipe } from 'angular-pipes/src/string/with.pipe';
 {{'The Flash Reverse' | with: null,null}} <!-- result: 'The Flash Reverse'-->
 {{'The Flash Reverse' | with}} <!-- result: 'The Flash Reverse'-->
 ```
+
+####reversestr
+
+Reverse a string.
```
Comment (fknop) [inline docs/string.md]: Add a note that this pipe will not work for special unicode character.

http://stackoverflow.com/questions/958908/how-do-you-reverse-a-string-in-place-in-javascript#959004

File: src/aggregate/group-by.pipe.ts [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+
+    transform(input: any, prop: string): Array<any> {
+
+        if (!isArray(input)) {
+            return input;
+        }
+
+        const arr: { [key: string]: Array<any> } = {};
+
+        for(const value of input) {
```
Comment (fknop) [inline src/aggregate/group-by.pipe.ts]: Add a space between `for` and `(`

File: src/string/reverse-str.pipe.spec.ts [hunk 3]
```diff
@@ -0,0 +1,33 @@
+import { ReverseStrPipe } from './reverse-str.pipe';
+
+
+
+describe('ReverseStrPipe', () => {
```
Comment (fknop) [inline src/string/reverse-str.pipe.spec.ts]: Missing a test where the input is not a string

File: src/string/reverse-str.pipe.ts [hunk 4]
```diff
@@ -0,0 +1,17 @@
...
+})
+export class ReverseStrPipe implements PipeTransform {
+
+    transform(input: string): string {
+
+        if (!isString(input) || isUndefined(input)) {
+            return input;
+        }
+
+		return input.split('').reverse().join('');
```
Comment (fknop) [inline src/string/reverse-str.pipe.ts]: Fix indentation

File: src/string/reverse-str.pipe.ts [hunk 5]
```diff
@@ -0,0 +1,17 @@
...
+import { isString, isUndefined } from '../utils/utils';
+
+@Pipe({
+	name: 'reverseStr'
+})
+export class ReverseStrPipe implements PipeTransform {
+
+    transform(input: string): string {
+
+        if (!isString(input) || isUndefined(input)) {
```
Comment (fknop) [inline src/string/reverse-str.pipe.ts]: No need to test if it's undefined. If it's undefined, it's not a string.

**Comment by rafaelss95** [comment]
I think it's everything fine now.

**Comment by fknop** [comment]
Publishing this in a few hours.

Thanks.

**Pull request merged**